Client/Frontend/Home/RecentTabsPanel.swift

```diff
@@ -0,0 +1,162 @@
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    func notificationReceived(notification: NSNotification) {
+        switch notification.name {
+        case NotificationPrivateDataClearedHistory:
+            profile.prefs.setObject(nil, forKey: "recentlyClosedTabs")
+            break
+        case NotificationDynamicFontChanged:
+            reloadData()
+            break
+        default:
+            // no need to do anything at all
+            log.warning("Received unexpected notification \(notification.name)")
+            break
+        }
+    }
+
+    override func reloadData() {
+        if let recentTabs = profile.prefs.dictionaryForKey("recentlyClosedTabs") {
+            recentlyClosedTabs = recentTabs as! [String:[String]]
+        }
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
+
+        if let cell = cell as? TwoLineTableViewCell {
+            cell.setLines(recentlyClosedTabs["titles"]![indexPath.row], detailText: recentlyClosedTabs["urls"]![indexPath.row])
+            cell.imageView?.sd_setImageWithURL((recentlyClosedTabs["faviconUrls"]![indexPath.row] ).asURL!) { (img, err, type, url) -> Void in
+                guard img != nil else {
+                    cell.imageView?.setIcon(nil, withPlaceholder: FaviconFetcher.getDefaultFavicon(NSURL(string: (self.recentlyClosedTabs["urls"]![indexPath.row]))!))
+                    return
+                }
+            }
+        }
+
+        return cell
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        if let url = NSURL(string: (recentlyClosedTabs["urls"]![indexPath.row])) {
+            let visitType = VisitType.Typed    // Means History, too.
+            if let homePanelDelegate = homePanelDelegate,
+                historyPanel = historyPanel {
+                homePanelDelegate.homePanel(historyPanel, didSelectURL: url, visitType: visitType)
+            }
+            return
+        }
+        log.warning("No site or no URL when selecting row.")
+    }
+
+    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 0
+    }
+
+    // Functions that deal with showing header rows.
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let count = recentlyClosedTabs["urls"]!.count
+        return count < 5 ? count : 5
```

Comment:
Please add a comment here explaining what this check is for. I'm having trouble figuring out what this `5` is supposed to represent, and what this code is supposed to be doing.